releasenotes/notes/add-classical-predecessors-9ecef0561822e934.yaml

```diff
@@ -0,0 +1,43 @@
+---
+features:
+  - |
+    Adds :meth:`qiskit.dagcircuit.DAGCircuit.classical_predecessors` and
+    :meth:`qiskit.dagcircuit.DAGCircuit.classical_successors`,an alternative to select the classical
+    wires without having to go to the inner graph object directly of a node in the DAG.
+    The following example illustrates the new functionality::
+
+        from qiskit import QuantumCircuit
+        from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+        from qiskit.converters import circuit_to_dag
+        from qiskit.circuit.library import RZGate
+
+        q = QuantumRegister(3, 'q')
+        c = ClassicalRegister(3, 'c')
+        circ = QuantumCircuit(q, c)
+        circ.h(q[0])
+        circ.cx(q[0], q[1])
+        circ.measure(q[0], c[0])
+        circ.rz(0.5, q[1]).c_if(c, 2)
+        circ.measure(q[1], c[0])
+        dag = circuit_to_dag(circ)
+
+        rz_node = dag.op_nodes(RZGate)[0]
+
+        print("Classical predecessors of RZ:")
+        for predecessor in dag.classical_predecessors(node):
+            print(type(predecessor))
+
+        print("Classical successors of RZ:")
+        for successors in dag.classical_successors(node):
+            print(type(successors))
+
+    This code results in the following output::
+
+        Classical predecessors of RZ:
+        <class 'qiskit.dagcircuit.dagnode.DAGInNode'>
+        <class 'qiskit.dagcircuit.dagnode.DAGInNode'>
+        <class 'qiskit.dagcircuit.dagnode.DAGOpNode'>
+        Classical successors of RZ:
+        <class 'qiskit.dagcircuit.dagnode.DAGOpNode'>
+        <class 'qiskit.dagcircuit.dagnode.DAGOutNode'>
+        <class 'qiskit.dagcircuit.dagnode.DAGOutNode'>
```

Comment:
yes, you are right with respect the first 2 points, the third point is correct is rz_node and for the fourth point I'm not sure to  eliminate the outputs, it could be great if we put a example with more gates and see the proposal solution

Reply:
I'm sorry, when I first posted it, I didn't see that Luciano had originally suggested printing the outputs.  I'm not actually sure I agree with that, because the output here isn't really super meaningful - these objects don't really have clear outputs, so printing them out doesn't give much information.

The thing with a release note is that it doesn't really need to include _all_ the information - that's what the full API documentation is for (the docstring).  The aim of an example in the release notes is usually just to illustrate the very high-level usage to make people aware of it.

Since Luciano originally suggested it, let's ask him (@1ucian0) whether he agrees, but my inclination would be to avoid outputs here, since they're not very useful.

Reply:
@jakelishman I did the changes, and Luciano is agree with your suggestion